File: carcassonne-master/src/tile_set.cc
#include "tile_set.h"

namespace carcassonne
{

TileSet::TileSet()
{
	ValidTiles valid_tiles_;

	starting_tile_begin_ = valid_tiles_.city_road_field_road_copy();
	
// TODO:Make a function that pushes a certain number of tiles into the tile 
//	 set void push_back_n(Tile**, int);
	bagable_tiles_.push_back(valid_tiles_.city_city_field_city_copy());
	bagable_tiles_.push_back(valid_tiles_.city_city_field_city_copy());
	bagable_tiles_.push_back(valid_tiles_.city_city_field_city_copy());
	
	bagable_tiles_.push_back(valid_tiles_.city_city_field_city_p_copy());
	
	bagable_tiles_.push_back(valid_tiles_.city_city_city_city_p_copy());
	
}

Tile& TileSet::starting_tile_begin()
{
	return starting_tile_begin_;
}

std::vector<Tile>& TileSet::bagable_tiles()
{
	return bagable_tiles_;
}

TileSet::~TileSet()
{

}



}

File: js/TileSet.js
import ValidTiles from "./ValidTiles"

export default class TileSet {
    constructor() {
        this.bagable_tiles_ = [];
        this.valid_tiles_ = new ValidTiles();
        this.starting_tile_begin_ = this.valid_tiles_.city_road_field_road_copy();
        // TODO:Make a function that pushes a certain number of tiles into the tile
//	 set void push_n(Tile**, int);
        this.bagable_tiles_.push(this.valid_tiles_.city_city_field_city_copy());
        this.bagable_tiles_.push(this.valid_tiles_.city_city_field_city_copy());
        this.bagable_tiles_.push(this.valid_tiles_.city_city_field_city_copy());

        this.bagable_tiles_.push(this.valid_tiles_.city_city_field_city_p_copy());

        this.bagable_tiles_.push(this.valid_tiles_.city_city_city_city_p_copy());
    }

    starting_tile_begin() {
        return this.starting_tile_begin_;
    }

    bagable_tiles() {
        return this.bagable_tiles_;
    }
}